src/main/java/org/opensearch/flowframework/transport/CreateWorkflowTransportAction.java

```diff
@@ -233,43 +235,80 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<Work
             );
         } else {
             // This is an existing workflow (PUT)
+            final boolean isFieldUpdate = request.isUpdateFields();
             // Fetch existing entry for time stamps
             logger.info("Querying existing workflow from global context: {}", workflowId);
             try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
                 client.get(new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId), ActionListener.wrap(getResponse -> {
                     context.restore();
                     if (getResponse.isExists()) {
                         Template existingTemplate = Template.parse(getResponse.getSourceAsString());
-                        // Update existing entry, full document replacement
-                        Template template = new Template.Builder(templateWithUser).createdTime(existingTemplate.createdTime())
-                            .lastUpdatedTime(Instant.now())
-                            .lastProvisionedTime(existingTemplate.lastProvisionedTime())
-                            .build();
+                        Template template;
+                        if (isFieldUpdate) {
+                            // Update only specified fields in existing
+                            Builder builder = new Template.Builder(existingTemplate).lastUpdatedTime(Instant.now());
+                            if (templateWithUser.name() != null) {
+                                builder.name(templateWithUser.name());
+                            }
+                            if (!Strings.isBlank(templateWithUser.description())) {
+                                builder.description(templateWithUser.description());
+                            }
+                            if (!Strings.isBlank(templateWithUser.useCase())) {
+                                builder.useCase(templateWithUser.useCase());
+                            }
+                            if (templateWithUser.templateVersion() != null) {
+                                builder.templateVersion(templateWithUser.templateVersion());
+                            }
+                            if (!templateWithUser.compatibilityVersion().isEmpty()) {
+                                builder.compatibilityVersion(templateWithUser.compatibilityVersion());
+                            }
+                            if (templateWithUser.getUiMetadata() != null) {
+                                builder.uiMetadata(templateWithUser.getUiMetadata());
+                            }
+                            template = builder.build();
```

Comment:
We should create a separate method for the builder here

Reply:
Not sure what you mean here.

Reply:
For[ L249-268](https://github.com/opensearch-project/flow-framework/pull/757/files#diff-167ff0aad0354b79b0a51b5713f3e951d9af59ded66c73f35cbece5f76ac90fcR249-R268), we should move them to a new method rather the current method bulky.

Reply:
I really don't see a way to do that generally. They are all conditional updates of fields in the builder.  At best I could just chain the builder setters like usually and internally handle null, but it's entirely possible that I would want to set some fields to null.  I could maybe iterate over a list of String field names and move those into a map in the builder that maintains the fields to be set instead of object fields but that just moves this relatively simple list of conditionals into ugly code implementation.

If you have an example of how you'd do this simpler I'd love to see it.

Reply:
Something like the below

```java
Template existingTemplate = Template.parse(getResponse.getSourceAsString());
Template template = buildTemplate(existingTemplate, templateWithUser, isFieldUpdate);
            
private Template buildTemplate(Template existingTemplate, Template templateWithUser, boolean isFieldUpdate) {
    if (isFieldUpdate) {
        // Update only specified fields in existing
        Builder builder = new Template.Builder(existingTemplate).lastUpdatedTime(Instant.now());
        updateBuilder(builder, templateWithUser);
        return builder.build();
    } else {
        // Update existing entry, full document replacement
        return new Template.Builder(templateWithUser)
                .createdTime(existingTemplate.createdTime())
                .lastUpdatedTime(Instant.now())
                .lastProvisionedTime(existingTemplate.lastProvisionedTime())
                .build();
    }
}

private void updateBuilder(Builder builder, Template templateWithUser) {
    if (templateWithUser.name() != null) {
        builder.name(templateWithUser.name());
    }
    if (!Strings.isBlank(templateWithUser.description())) {
        builder.description(templateWithUser.description());
    }
    if (!Strings.isBlank(templateWithUser.useCase())) {
        builder.useCase(templateWithUser.useCase());
    }
    if (templateWithUser.templateVersion() != null) {
        builder.templateVersion(templateWithUser.templateVersion());
    }
    if (!templateWithUser.compatibilityVersion().isEmpty()) {
        builder.compatibilityVersion(templateWithUser.compatibilityVersion());
    }
    if (templateWithUser.getUiMetadata() != null) {
        builder.uiMetadata(templateWithUser.getUiMetadata());
    }
}
```

Reply:
@owaiskazi19 d0672728054ba702ca4ddd67e386f9b889707113